src/lib.rs

```diff
@@ -283,6 +283,12 @@ impl<const BITS: usize, const LIMBS: usize> Uint<BITS, LIMBS> {
             (_, true) => Self::MAX,
         }
     }
+
+    /// Check if this uint is zero
+    #[must_use]
+    pub fn is_zero(&self) -> bool {
```

Comment:
This should be moved out of `lib.rs`, which is mostly constructors and low-level limb stuff (which this is not).

`cmp.rs` would be a good place.